web-ui/src/main/resources/catalog/components/common/map/mapService.js

```diff
@@ -1432,19 +1442,41 @@
             return defer.promise;
           },
 
+          /**
+           * @ngdoc method
+           * @methodOf gn_map.addEsriRestFromScratch:gnMap
+           * @name gnMap#addEsriRestFromScratch
+           *
+           * @description
+           * Here is the method to use when you want to add a ESRIREst layer from
+           * a url and a name (layer identifier).
+           *
+           * Return a promise with ol.Layer as data is succeed, and url/name
+           * if failure.
+           * If createOnly, we don't add the layer to the map.
+           * If the md object is given, we add it to the layer, or we try
+           * to retrieve it in the catalog
+           *
+           * @param {ol.Map} map to add the layer
+           * @param {string} url of the service
+           * @param {string} name of the layer (identifier)
+           * @param {boolean} createOnly or add it to the map
+           * @param {!Object} md object
+           */
           addEsriRestFromScratch: function(map, url, name, createOnly, md) {
             if (url === '') {
               var error = "Trying to add an ESRI layer with no service URL. Layer name is " + name + ". Check the metadata or the map.";
               console.warn(error);
               return $q.reject(error);
             }
             var serviceUrl = url.replace(/(.*\/MapServer).*/, '$1');
-            var layer = !!name && parseInt(name).toString() === name
+            var layer = angular.isNumber(name)
```

Comment:
That does not work with the comment block above, which is stating that `name` is a string

Reply:
name is not a string, at least from my tests. Check https://land.discomap.eea.europa.eu/arcgis/rest/services/Agriculture/agri_extensification_on_HNVfarmland06/MapServer?f=pjson

The `name` is taken from the layer ids, also id=0 is valid, and caused issues with `!!name` that was evaluated to `false`

```
"layers": [
  {
   "id": 0,
```